### PR TITLE
Bootstrap Site profiler

### LINKED ISSUE #81473: Bootstrap `/site-profiler` calypso app

### PR DESCRIPTION
**Pull request by bogiii**
Closes #81473

## Proposed Changes

* Introduce `site-profiler` feature flag (development env for now)
* Bootstrap a `SiteProfiler` mini app accessible on the `/site-profiler` route

## Testing Instructions

* Go to `/site-profiler` (for non-development env, it will redirect to the root `/`)
* Check if there is a "DomainAnalyzer" block rendered there

* NOTE: the form doesn't have any logic so far; in the next iteration, the focus will be the concrete flow

**Non-logged in user**
Screenshot 2023-09-14 at 14 12 13

**Logged in user**
Screenshot 2023-09-14 at 14 11 51

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/data/site-profiler/use-domain-analyzer-query.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export interface MigrationStatusError {
+	status: number;
+	message: string;
+}
+
+export const useDomainAnalyzerQuery = ( domain: string ) => {
+	return useQuery( {
+		queryKey: [ 'domain-', domain ],
+		queryFn: (): Promise< SourceSiteMigrationDetails > =>
```
Comment (zaerl) [inline client/data/site-profiler/use-domain-analyzer-query.ts]: We can create a base response type for this first bootstrap. This one is enough for now.

```ts
{
  domain: string;
  hosting_provider: {
    name: string;
  };
  whois: any;
  dns: any;
}
```
We can then create a more detailed one.

**Review by zaerl — CHANGES_REQUESTED** [review]
Left a comment for you. Let me know what you think.

File: client/data/site-profiler/use-domain-analyzer-query.ts (hunk 1 above)
Comment (bogiii) [reply client/data/site-profiler/use-domain-analyzer-query.ts]: Good catch, this is copy/paste (SourceSiteMigrationDetails) momentum, and I forgot to update the response type.

**Comment by vishnugopal** [comment]
@bogiii  Not for this PR, but I think we definitely need to do something about this loading flicker. Especially when you are logged in, this is very jarring. We should think about having proper SSR here.

https://github.com/Automattic/wp-calypso/assets/533/851ac06f-ec9e-447a-b616-20267791d7b8

File: client/site-profiler/components/domain-analyzer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+			<p>
+				{ translate(
+					'Access essential information about a site, such as hosting provider, domain details, and contact information.'
+				) }
+			</p>
+
+			<form className="domain-analyzer--form" onSubmit={ onSubmit }>
+				<div className="domain-analyzer--form-container">
+					<div className="col-1">
+						<input type="text" name="domain" placeholder={ translate( 'mysite.com' ) } />
```
Comment (vishnugopal) [inline client/site-profiler/components/domain-analyzer/index.tsx:29]: Small nit: can we add `autocomplete="off"` here? Thanks!

**Review by zaerl — APPROVED** [review]
LGTM.

File: client/site-profiler/components/domain-analyzer/index.tsx (hunk 2 above)
Comment (bogiii) [reply client/site-profiler/components/domain-analyzer/index.tsx:29]: For sure. Thanks!

**Comment by bogiii** [comment]
> but I think we definitely need to do something about this loading flicker. Especially when you are logged in, this is very jarring. We should think about having proper SSR here.

@vishnugopal, I absolutelly agree with you.
I will see to cover it in the next iterations! Thanks!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8941739

Thank you @bogiii for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8941739

Thank you @bogiii for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8941739).

**Pull request merged**